business/checkers/no_service_checker.go

```diff
@@ -30,12 +30,12 @@ func (in NoServiceChecker) Check() models.IstioValidations {
 		return validations
 	}
 
-	serviceNames := in.ServiceList.GetServiceNames()
 	serviceHosts := kubernetes.ServiceEntryHostnames(append(in.IstioConfigList.ServiceEntries, in.ExportedResources.ServiceEntries...))
 	gatewayNames := kubernetes.GatewayNames(in.GatewaysPerNamespace)
 
 	for _, virtualService := range in.IstioConfigList.VirtualServices {
-		validations.MergeValidations(runVirtualServiceCheck(virtualService, in.Namespace, serviceNames, serviceHosts, in.Namespaces, in.RegistryServices))
+		validations.MergeValidations(runVirtualServiceCheck(virtualService, in.Namespace, in.ServiceList, serviceHosts, in.Namespaces, in.RegistryServices))
```

Comment:
Perhaps this should need a more deep refactoring if we start using the registry ?

I see that ServiceEntries were also added to this validations:
![image](https://user-images.githubusercontent.com/1662329/144223584-4c9392bc-291b-48a3-8ced-2a3bbacf1a56.png)

But now those should be available in the ServiceList with the AllNamespaces flag.

I'm just asking, I know this is part of the https://github.com/kiali/kiali/issues/4528 and you may want to plan to add it on that effort (and it's me mixing both contexts during this PR review).

Reply:
As I understand the context, I'm fine if we want to consolidate this work and waiting for #4528

Reply:
Ok, now I understand the context, validations were using the RegistryServices as the last way to check the exported services.

Yes, it makes sense in the context of the PR, I think we can consolidate it and wait for #4528 to unify to only use a single source for fetching the Services.

I was confused with the introduction of the AllNamespace flag in this PR.

Reply:
I think the ServiceEntries from Registry could be added in a followup PRs.

Reply:
> I think the ServiceEntries from Registry could be added in a followup PRs.

Well, I think it's part of the #4528 effort, to revisit the information needed in the business/istio_validations.go to propagate to the checkers.

There will be more cases like this one.